Analysis/Tasks/PWGHF/HFJpsiToEECandidateSelector.cxx

```diff
@@ -0,0 +1,235 @@
+      return false; //is it clusters findable or found - need to check
+    }
+    return true;
+  }
+
+  /// Conjugate independent toplogical cuts
+  /// \param hfCandProng2 is candidate
+  /// \param trackPositron is the track with the positron hypothesis
+  /// \param trackElectron is the track with the electron hypothesis
+  /// \return true if candidate passes all cuts
+  template <typename T1, typename T2>
+  bool selectionTopol(const T1& hfCandProng2, const T2& trackPositron, const T2& trackElectron)
+  {
+    auto candpT = hfCandProng2.pt();
+    int pTBin = getpTBin(candpT);
+    if (pTBin == -1) {
+      return false;
+    }
+
+    if (candpT < d_pTCandMin || candpT >= d_pTCandMax) {
+      return false; //check that the candidate pT is within the analysis range
+    }
+
+    if (TMath::Abs(InvMassJpsi(hfCandProng2) - RecoDecay::getMassPDG(443)) > cuts[pTBin][0]) {
+      return false;
+    }
+
+    if ((trackElectron.pt() < cuts[pTBin][3]) || (trackPositron.pt() < cuts[pTBin][3])) {
+      return false; //cut on daughter pT
+    }
+    if (TMath::Abs(trackElectron.dcaPrim0()) > cuts[pTBin][1] || TMath::Abs(trackPositron.dcaPrim0()) > cuts[pTBin][1]) {
+      return false; //cut on daughter dca - need to add secondary vertex constraint here
+    }
+    if (TMath::Abs(trackElectron.dcaPrim1()) > cuts[pTBin][2] || TMath::Abs(trackPositron.dcaPrim1()) > cuts[pTBin][2]) {
+      return false; //cut on daughter dca - need to add secondary vertex constraint here
+    }
+
+    return true;
+  }
+
+  /// Check if track is ok for TPC PID
+  /// \param track is the track
+  /// \note function to be expanded
+  /// \return true if track is ok for TPC PID
+  template <typename T>
+  bool validTPCPID(const T& track)
+  {
+    if (TMath::Abs(track.pt()) < d_pidTPCMinpT || TMath::Abs(track.pt()) >= d_pidTPCMaxpT) {
+      return false;
+    }
+    //if (track.TPCNClsFindable() < d_TPCNClsFindablePIDCut) return false;
+    return true;
+  }
+
+  /// Check if track is compatible with given TPC Nsigma cut for a given flavour hypothesis
+  /// \param track is the track
+  /// \param nPDG is the flavour hypothesis PDG number
+  /// \param nSigmaCut is the nsigma threshold to test against
+  /// \note nPDG=11 electron
+  /// \return true if track satisfies TPC PID hypothesis for given Nsigma cut
+  template <typename T>
+  bool selectionPIDTPC(const T& track, int nSigmaCut)
+  {
+    double nSigma = 1.0; //arbitarily large value
+    nPDG = TMath::Abs(11);
+    //if (nPDG == 11) {
+    //  nSigma = track.tpcNSigmaEl();
+    //} else {
+    //  return false;
+    //}
+    return nSigma < nSigmaCut;
+  }
+
+  /// PID selection on daughter track
+  /// \param track is the daughter track
+  /// \param nPDG is the PDG code of the flavour hypothesis
+  /// \note nPDG=11 electron
+  /// \return 1 if successful PID match, 0 if successful PID rejection, -1 if no PID info
+  template <typename T>
+  int selectionPID(const T& track, int nPDG)
```

Comment:
If only decays to electrons are meant to be considered in this selector, this part should eventually look something like:
```c++
  /// Check if track is compatible with given TPC Nsigma cut
  /// \param track is the track
  /// \param nSigmaCut is the nsigma threshold to test against
  /// \return true if track satisfies TPC PID hypothesis for given Nsigma cut
  template <typename T>
  bool selectionPIDTPC(const T& track, int nSigmaCut)
  {
    return true; // FIXME: temporarily disabled
    //return track.tpcNSigmaEl() < nSigmaCut;
  }

  /// PID selection on daughter track
  /// \param track is the daughter track
  /// \return 1 if successful PID match, 0 if successful PID rejection, -1 if no PID info
  template <typename T>
  int selectionPID(const T& track)
  {
```